File: app/src/main/java/ca/mudar/rotationquicksetting/data/UserPrefs.java
package ca.mudar.rotationquicksetting.data;

import android.content.Context;
import android.content.SharedPreferences;
import android.preference.PreferenceManager;

import ca.mudar.rotationquicksetting.Const;
import ca.mudar.rotationquicksetting.Const.PrefsNames;
import ca.mudar.rotationquicksetting.R;

import static ca.mudar.rotationquicksetting.Const.PrefsNames.HAS_HELP;
import static ca.mudar.rotationquicksetting.Const.PrefsNames.HAS_ONBOARDING;

/**
 * Created by mudar on 10/05/17.
 */

public class UserPrefs {

    private static UserPrefs instance;
    private SharedPreferences mPrefs;
    private SharedPreferences.Editor mPrefsEditor;

    private UserPrefs(Context context) {
        mPrefs = context.getSharedPreferences(Const.APP_PREFS_NAME, Context.MODE_PRIVATE);
    }

    public static UserPrefs getInstance(Context context) {
        if (instance == null) {
            instance = new UserPrefs(context);
        }
        return instance;
    }

    private SharedPreferences.Editor edit() {
        if (mPrefsEditor == null) {
            mPrefsEditor = mPrefs.edit();
        }

        return mPrefsEditor;
    }

    public static void setDefaults(Context context) {
        PreferenceManager.setDefaultValues(context,
                Const.APP_PREFS_NAME,
                Context.MODE_PRIVATE,
                R.xml.prefs_defaults,
                false);
    }


    public boolean hasOnboarding() {
        return mPrefs.getBoolean(HAS_ONBOARDING, true);
    }

    public void setOnboardingCompleted() {
        edit().putBoolean(HAS_ONBOARDING, false)
                .commit();
    }

    public boolean hasHelp() {
        return mPrefs.getBoolean(HAS_HELP, true);
    }

    public void setHelpCompleted() {
        edit().putBoolean(HAS_HELP, false)
                .commit();
    }

    public boolean isReversePortrait() {
        return mPrefs.getBoolean(PrefsNames.REVERSE_PORT, false);
    }

    public boolean isReverseLandscape() {
        return mPrefs.getBoolean(PrefsNames.REVERSE_LAND, false);
    }

    public boolean hasAutoRotate() {
        return mPrefs.getBoolean(PrefsNames.AUTO_ROTATE, false);
    }
}

File: app/src/main/java/ca/mudar/rotationquicksetting/data/UserPrefs.kt
/*
 * Rotation Quick Setting
 * Add a Quick Settings tile to select portrait or landscape screen orientation.
 *
 * Copyright (C) 2017 Mudar Noufal <mn@mudar.ca>
 *
 * This file is part of RotationQS.
 *
 * This program is free software: you can redistribute it and/or modify
 * it under the terms of the GNU General Public License as published by
 * the Free Software Foundation, either version 3 of the License, or
 * (at your option) any later version.
 *
 * This program is distributed in the hope that it will be useful,
 * but WITHOUT ANY WARRANTY; without even the implied warranty of
 * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
 * GNU General Public License for more details.
 *
 * You should have received a copy of the GNU General Public License
 * along with this program.  If not, see <http://www.gnu.org/licenses/>.
 */

package ca.mudar.rotationquicksetting.data

import android.content.Context
import android.content.SharedPreferences
import android.preference.PreferenceManager
import ca.mudar.rotationquicksetting.Const
import ca.mudar.rotationquicksetting.Const.PrefsNames
import ca.mudar.rotationquicksetting.Const.PrefsNames.HAS_HELP
import ca.mudar.rotationquicksetting.Const.PrefsNames.HAS_ONBOARDING
import ca.mudar.rotationquicksetting.R

class UserPrefs constructor(context: Context) {

    companion object {
        fun setDefaults(context: Context) {
            PreferenceManager.setDefaultValues(context,
                    Const.APP_PREFS_NAME,
                    Context.MODE_PRIVATE,
                    R.xml.prefs_defaults,
                    false)
        }
    }

    private val prefs = context.getSharedPreferences(Const.APP_PREFS_NAME, Context.MODE_PRIVATE)
    private var prefsEditor: SharedPreferences.Editor? = null

    val isReversePortrait: Boolean
        get() = prefs.getBoolean(PrefsNames.REVERSE_PORT, false)

    val isReverseLandscape: Boolean
        get() = prefs.getBoolean(PrefsNames.REVERSE_LAND, false)

    private fun edit(): SharedPreferences.Editor {
        if (prefsEditor == null) {
            prefsEditor = prefs.edit()
        }

        return prefsEditor!!
    }


    fun hasOnboarding(): Boolean {
        return prefs.getBoolean(HAS_ONBOARDING, true)
    }

    fun setOnboardingCompleted() {
        edit().putBoolean(HAS_ONBOARDING, false)
                .commit()
    }

    fun hasHelp(): Boolean {
        return prefs.getBoolean(HAS_HELP, true)
    }

    fun setHelpCompleted() {
        edit().putBoolean(HAS_HELP, false)
                .commit()
    }

    fun hasAutoRotate(): Boolean {
        return prefs.getBoolean(PrefsNames.AUTO_ROTATE, false)
    }
}
